### PR TITLE
Feat/create a task

### PR DESCRIPTION
**Pull request by KivancUlucan**

### CODE REVIEW

**Review by doctype-Melvin — CHANGES_REQUESTED** [review]
Hi Kivanc!

You're doing great. 

I left a couple suggestions for you to consider when revising your code. Keep it DRY (don't repeat yourself) would be my biggest tip.

Keep it going!

File: pages/addTask.js [hunk 1]
```diff
@@ -1,3 +1,151 @@
...
+  display: flex;
+  flex-direction: column;
+  margin-left: 3rem;
+`;
+const LabelForTitle = styled.label`
+  display: flex;
+  flex-direction: column;
+  font-size: 0.8rem;
+  margin-bottom: 0.2rem;
+`;
```
Comment (doctype-Melvin) [inline pages/addTask.js:72]:
```suggestion
const StyledLabel = styled.label`
  display: flex;
  flex-direction: column;
  font-size: 0.8rem;
  margin-bottom: 0.2rem;
`;
```

File: pages/addTask.js [hunk 2]
```diff
@@ -1,3 +1,151 @@
...
+const InputForTitle = styled.input`
+  border: 0.03rem solid #d1d1d1;
+  border-radius: 0.3rem;
+  width: 20rem;
+  height: 2.5rem;
+  outline: none;
+  padding: 0.5rem 1rem;
+  margin-bottom: 1rem;
+`;
```
Comment (doctype-Melvin) [inline pages/addTask.js:81]:
```suggestion
const StyledInput = styled.input`
  border: 0.03rem solid #d1d1d1;
  border-radius: 0.3rem;
  width: 20rem;
  height: 2.5rem;
  outline: none;
  padding: 0.5rem 1rem;
  margin-bottom: 1rem;
`;
```

File: pages/addTask.js [hunk 3]
```diff
@@ -1,3 +1,151 @@
...
+const LabelForDescription = styled.label`
```
Comment (doctype-Melvin) [inline pages/addTask.js:82]:
```suggestion
const StyledLabel = styled.label`
```

File: pages/addTask.js [hunk 4]
```diff
@@ -1,3 +1,151 @@
...
+const TextareaForDescription = styled.textarea`
+  border: 0.03rem solid #d1d1d1;
+  border-radius: 0.3rem;
+  width: 20rem;
+  height: 2.5rem;
+  outline: none;
+  padding: 0.5rem 1rem;
+  margin-bottom: 1rem;
+`;
+const LabelForPriority = styled.label`
```
Comment (doctype-Melvin) [inline pages/addTask.js:97]:
```suggestion
const StyledLabel = styled.label`
```

File: pages/addTask.js [hunk 5]
```diff
@@ -1,3 +1,151 @@
...
+const InputForDueDate = styled.input`
+  border: 0.03rem solid #d1d1d1;
+  border-radius: 0.3rem;
+  width: 20rem;
+  height: 2.5rem;
+  outline: none;
+  padding: 0.5rem 1rem;
+  margin-bottom: 1rem;
+`;
+const LabelForCategory = styled.label`
```
Comment (doctype-Melvin) [inline pages/addTask.js:119]:
```suggestion
const StyledLabel = styled.label`
```

File: pages/addTask.js [hunk 6]
```diff
@@ -1,3 +1,151 @@
...
+`;
+const SelectForPriority = styled.select`
+  border: 0.03rem solid #d1d1d1;
+  border-radius: 0.3rem;
+  width: 20rem;
+  height: 2.5rem;
+  outline: none;
+  padding: 0.5rem 1rem;
+  margin-bottom: 1rem;
+`;
```
Comment (doctype-Melvin) [inline pages/addTask.js:133]: This has the same styling rules as SelectForCategory. StyledSelect would make sense here.

**Review by doctype-Melvin — CHANGES_REQUESTED** [review]
Hi Kivanc,

I've left you some suggestions for your code revision. You could make your life easier if you keep your code DRY (don't repeat yourself). 

Other than that, you're doing great! Keep it going!

File: components/TaskCard/TaskCard.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import styled from "styled-components";
+
+export default function TaskCard({ task }) {
+  return (
+    <Card>
+      <Title>Title: {task.title}</Title>
+      <Description>Description:{task.description} </Description>
```
Comment (djfarly) [inline components/TaskCard/TaskCard.js:7]: misaligned space
```suggestion
      <Description>Description: {task.description}</Description>
```

File: pages/board.js [hunk 8]
```diff
@@ -1,3 +1,20 @@
...
+import TaskCard from "../components/TaskCard/TaskCard";
+import styled from "styled-components";
+
+export default function Board({ tasks }) {
+  console.log(tasks);
+  return (
+    <>
+      <CardContainer>
+        {tasks.map((task) => (
+          <TaskCard key={task.title} task={task} />
```
Comment (djfarly) [inline pages/board.js:11]: the task title might not be the best candidate for `key`. Keys have to be unique per list, so if you have two tasks with the same title React will rightfully complain.

Consider adding an `id` to your tasks to have a unique identifier.

To do so install `uid`

```
npm i uid
```

and then change your `handleAddTask` function in `_app.js` to create ids:

```js
import { uid } from "uid";

/* … */

  function handleAddTask(newTask) {
    setTasks([...tasks, {...newTask, id: uid()}]);
  }
```

then you can do:
```suggestion
          <TaskCard key={task.id} task={task} />
```

File: pages/board.js [hunk 9]
```diff
@@ -1,3 +1,20 @@
-export default function BoardPage() {
-  return <h1>Board</h1>;
+import { useState } from "react";
+import TaskCard from "../components/TaskCard/TaskCard";
+import styled from "styled-components";
+
+export default function Board({ tasks }) {
+  console.log(tasks);
+  return (
+    <>
```
Comment (djfarly) [inline pages/board.js:8]: you do not need this fragment

File: pages/board.js [hunk 10]
```diff
@@ -1,3 +1,20 @@
-export default function BoardPage() {
-  return <h1>Board</h1>;
+import { useState } from "react";
+import TaskCard from "../components/TaskCard/TaskCard";
+import styled from "styled-components";
+
+export default function Board({ tasks }) {
+  console.log(tasks);
```
Comment (djfarly) [inline pages/board.js:6]: try not to merge debug console.logs

```suggestion
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Kivancm

thank you for the PR!

I only have one slightly larger comment (the one about adding an `id`), otherwise this looks very good to me!

Cheers, Jan

**Review by djfarly — APPROVED** [review]
👍 lgtm!

**Pull request merged**